### PR TITLE
fix: spanner integration test conflict resolution

### PR DESCRIPTION
**Pull request by hemanshv**
This resolves the issue https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/2021

### CODE REVIEW

**Comment by hemanshv** [comment]
Hi @amanda-tarafa ,
Opened this PR for internal lex review. 
Will let you know once the internal review get competed.

Thanks

**Comment by Rishabh-V** [comment]
I 'll have a look at it tomorrow morning. Thanks.

**Review by Rishabh-V — COMMENTED** [review]
Apologies for the delay. Just had a look. I will have another look after the current comments are addressed/responded. Thanks.

File: spanner/api/Spanner.Samples.Tests/QueryNewColumnAsyncTest.cs [hunk 1]
```diff
@@ -28,10 +28,15 @@ public QueryNewColumnAsyncTest(SpannerFixture spannerFixture)
...
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            await _spannerFixture.InitializeTempDatabaseAsync(databaseId);
+            QueryNewColumnAsyncSample sample = new QueryNewColumnAsyncSample();
+            await _spannerFixture.RefillMarketingBudgetsForTempDatabaseAsync(databaseId, 100000, 500000);
+
+            var albums = await sample.QueryNewColumnAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Contains(albums, a => a.AlbumId == 1 && a.MarketingBudget == 100000);
+            Assert.Contains(albums, a => a.AlbumId == 2 && a.MarketingBudget == 500000);
+        }, _spannerFixture.CreateSingersTableStatement, _spannerFixture.CreateAlbumsTableStatement);
```
[the remark this answers is not in the document]
Comment (hemanshv) [reply spanner/api/Spanner.Samples.Tests/QueryNewColumnAsyncTest.cs]: Lets keep it this way as I think this way is simple and easy to extend in future.

File: spanner/api/Spanner.Samples.Tests/QueryDataWithTimestampColumnAsyncTest.cs [hunk 2]
```diff
@@ -28,11 +28,20 @@ public QueryDataWithTimestampColumnAsyncTest(SpannerFixture spannerFixture)
...
+            AddCommitTimestampAsyncSample addCommitTimestampAsyncSample = new AddCommitTimestampAsyncSample();
+            await addCommitTimestampAsyncSample.AddCommitTimestampAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            QueryDataWithTimestampColumnAsyncSample sample = new QueryDataWithTimestampColumnAsyncSample();
+            UpdateDataWithTimestampColumnAsyncSample updateWithTimestamp = new UpdateDataWithTimestampColumnAsyncSample();
+            await updateWithTimestamp.UpdateDataWithTimestampColumnAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            var albums = await sample.QueryDataWithTimestampColumnAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Contains(albums, a => a.AlbumId == 1 && a.SingerId == 1 && a.LastUpdateTime != null);
+            Assert.Contains(albums, a => a.AlbumId == 2 && a.SingerId == 2 && a.LastUpdateTime != null);
+
```
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply spanner/api/Spanner.Samples.Tests/QueryDataWithTimestampColumnAsyncTest.cs:43]: ping on this.

File: spanner/api/Spanner.Samples.Tests/SpannerFixture.cs [hunk 3]
```diff
@@ -74,7 +74,21 @@ public class SpannerFixture : IAsyncLifetime, ICollectionFixture<SpannerFixture>
     public InstanceAdminClient InstanceAdminClient { get; private set; }
     public SpannerConnection AdminSpannerConnection { get; private set; }
     public SpannerConnection SpannerConnection { get; private set; }
-    public SpannerConnection PgSpannerConnection { get; private set; }
+    public SpannerConnection PgSpannerConnection { get; private set; }
+
+    public const string CreateSingersTableStatement =
```
Comment (Rishabh-V) [inline spanner/api/Spanner.Samples.Tests/SpannerFixture.cs:79]: nit: Can we please place constants at the top?

File: spanner/api/Spanner.Samples.Tests/SpannerFixture.cs [hunk 4]
```diff
@@ -514,11 +536,12 @@ public async Task CreateVenuesTableAndInsertDataAsync(string databaseId)
         }));
     }
 
-    public async Task RefillMarketingBudgetsAsync(int firstAlbumBudget, int secondAlbumBudget)
+    public async Task RefillMarketingBudgetsAsync(int firstAlbumBudget, int secondAlbumBudget, string databaseId = null)
     {
+        var spannerConnection = (databaseId == null) ? SpannerConnection : new SpannerConnection($"Data Source=projects/{ProjectId}/instances/{InstanceId}/databases/{databaseId}");
```
Comment (Rishabh-V) [inline spanner/api/Spanner.Samples.Tests/SpannerFixture.cs:541]:
```suggestion
        var spannerConnection = databaseId is null ? SpannerConnection : new SpannerConnection($"Data Source=projects/{ProjectId}/instances/{InstanceId}/databases/{databaseId}");
```
I don't think we need braces here.

File: spanner/api/Spanner.Samples.Tests/UpdateDataWithTimestampColumnTest.cs [hunk 5]
```diff
@@ -28,8 +28,17 @@ public UpdateDataWithTimestampColumnTest(SpannerFixture spannerFixture)
...
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            await _spannerFixture.InitializeTempDatabaseAsync(databaseId);
+            AddCommitTimestampAsyncSample addCommitTimestampAsyncSample = new AddCommitTimestampAsyncSample();
+            await addCommitTimestampAsyncSample.AddCommitTimestampAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            UpdateDataWithTimestampColumnAsyncSample sample = new UpdateDataWithTimestampColumnAsyncSample();
+            var rowCount = await sample.UpdateDataWithTimestampColumnAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Equal(2, rowCount);
+
```
Comment (Rishabh-V) [inline spanner/api/Spanner.Samples.Tests/UpdateDataWithTimestampColumnTest.cs:40]: nit: unnecessary line break here and at a few other places across files.

**Review by Rishabh-V — COMMENTED** [review]
Line breaks are not fixed across the files. Please address/respond to the comments and then request review from Amanda. Thanks.

**Comment by hemanshv** [comment]
Hi @amanda-tarafa , 
We are done with the internal review on this one. Please have a look and share your feedback.

Thanks,

**Comment by amanda-tarafa** [comment]
Please fix the conflicts so I can review, thanks!

**Comment by hemanshv** [comment]
Hey @amanda-tarafa ,
The conflict is now resolved. Please review and provide your feedback.

Thanks

File: spanner/api/Spanner.Samples.Tests/ReadStaleDataAsyncTest.cs [hunk 6]
```diff
@@ -29,14 +29,15 @@ public ReadStaleDataAsyncTest(SpannerFixture spannerFixture)
...
     public async Task TestReadStaleDataAsync()
     {
-        ReadStaleDataAsyncSample sample = new ReadStaleDataAsyncSample();
-        await _spannerFixture.RefillMarketingBudgetsAsync(300000, 300000);
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            await _spannerFixture.InitializeTempDatabaseAsync(databaseId);
+            ReadStaleDataAsyncSample sample = new ReadStaleDataAsyncSample();
+            await _spannerFixture.RefillMarketingBudgetsAsync(300000, 300000, databaseId);
+            await Task.Delay(TimeSpan.FromSeconds(15));
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/ReadStaleDataAsyncTest.cs:37]: Leave a note here saying that we need the wait to actually make the data stale.

File: spanner/api/Spanner.Samples.Tests/UpdateUsingDmlWithTimestampCoreAsyncTest.cs [hunk 7]
```diff
@@ -28,8 +28,15 @@ public UpdateUsingDmlWithTimestampCoreAsyncTest(SpannerFixture spannerFixture)
...
-        Assert.Equal(3, rowCount);
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            await _spannerFixture.InitializeTempDatabaseAsync(databaseId);
+            AddCommitTimestampAsyncSample addCommitTimestampAsyncSample = new AddCommitTimestampAsyncSample();
+            await addCommitTimestampAsyncSample.AddCommitTimestampAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            UpdateUsingDmlWithTimestampCoreAsyncSample sample = new UpdateUsingDmlWithTimestampCoreAsyncSample();
+            var rowCount = await sample.UpdateUsingDmlWithTimestampCoreAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Equal(2, rowCount);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/UpdateUsingDmlWithTimestampCoreAsyncTest.cs:39]: Why did the assert changed here?

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
One tiny change and a question, but this looks good. Thanks for doing this.

File: spanner/api/Spanner.Samples.Tests/UpdateUsingDmlWithTimestampCoreAsyncTest.cs (hunk 7 above)
Comment (hemanshv) [reply spanner/api/Spanner.Samples.Tests/UpdateUsingDmlWithTimestampCoreAsyncTest.cs:39]: Earlier UpdateUsingBatchDmlCoreAsync.cs used to update the Albums tables and insert a new row `"INSERT INTO Albums (SingerId, AlbumId, AlbumTitle, MarketingBudget) VALUES (1, 3, 'Test Album Title', 10000)"` which makes the count to 3. Now as they are independent its just 2 the one which is inserted by init function.

**Pull request merged**